wsi_deid/import_export.py

```diff
@@ -110,9 +110,32 @@ def getSchema():
     # that contain a single file that parses as valid json, then we need to
     # load those files and combine them into a single schema in lieu of using
     # this default SCHEMA_FILE_PATH
-    logger.info('Using %s (length %d) for schema' % (
-        SCHEMA_FILE_PATH, os.path.getsize(SCHEMA_FILE_PATH)))
-    return json.load(open(SCHEMA_FILE_PATH))
+    schemaFolderId = Setting().get(PluginSettings.WSI_DEID_SCHEMA_FOLDER)
+    mergedSchema = {}
+
+    if schemaFolderId:
+        schemaFolder = Folder().load(schemaFolderId, force=True)
+
+        for filepath, file in Folder().fileList(schemaFolder, data=False):
+            currentObject=json.load(File().open(file))
+            print(type(currentObject))
+            for key, value in currentObject.items():
+                if key in mergedSchema:
+                    if value != mergedSchema[key]:
+                        if isinstance(mergedSchema[key], list):
+                            mergedSchema[key].append(value)
+                        else:
+                            temp_list = [mergedSchema[key]]
+                            temp_list.append(value)
+                            mergedSchema[key]=temp_list
+
+                else:
+                    mergedSchema[key]=value
+    # logger.info('Using %s (length %d) for schema' % (
+    #     SCHEMA_FILE_PATH, os.path.getsize(SCHEMA_FILE_PATH)))
```

Comment:
If we didn't find any schemas in the schema folder (because the folder is unset, missing, or empty of json files), then we want to load the original SCHEMA_FILE_PATH as a fallback.

Reply:
added in 2a93feaf7fb521edcaeb54d9812d4320d10df69e